packages/flutter_tools/README.md

```diff
@@ -1,13 +1,36 @@
 # Flutter Tools
 
-Developer tools for building Flutter applications.
+Command line developer tools for building Flutter applications.
 
-Be sure to follow the instructions on
-[CONTRIBUTING.md](../../CONTRIBUTING.md) to setup.
+## Working on Flutter Tools
 
-To run the tests, ensure that no devices are connected,
-then navigate to `flutter_tools` and execute:
+Be sure to follow the instructions on [CONTRIBUTING.md](../../CONTRIBUTING.md)
+to set up your development environment.
 
+### Running
+
+To run Flutter Tools from source, in this directory run:
+```shell
+$ ../../bin/cache/dart-sdk/bin/dart bin/flutter_tools.dart
+```
+followed by command line arguments, as usual.
+
+### Analyzing
+
+To run the analyzer on Flutter Tools, in this directory run:
+```shell
+$ ../../bin/cache/dart-sdk/bin/dartanalyzer .
+```
+
+### Testing
+
+To run the tests in the `test/` directory, first ensure that there are no
```

Comment:
To run all tests precompiled, first set `FLUTTER_ROOT` to an appropriate value. then run `../../bin/cache/dart-sdk/bin/pub run build_runner test`

Reply:
Maybe also a note that the integration/command shard is really slow currently

Reply:
Added.